gctscript/modules/gct/common.go

```diff
@@ -0,0 +1,408 @@
+package gct
+
+import (
+	"errors"
+	"fmt"
+
+	objects "github.com/d5/tengo/v2"
+	"github.com/thrasher-corp/gocryptotrader/common/file"
+	"github.com/thrasher-corp/gocryptotrader/gctscript/modules/ta/indicators"
+)
+
+var commonModule = map[string]objects.Object{
+	"writeascsv": &objects.UserFunction{Name: "writeascsv", Value: WriteAsCSV},
+}
+
+// WriteAsCSV takes in a slice matrix to save to file
+func WriteAsCSV(args ...objects.Object) (objects.Object, error) {
+	if len(args) == 0 {
+		return nil, errors.New("cannot write to file, no data present")
+	}
+
+	var bucket [][]string
+	var err error
+	var target string
+	for i := range args {
+		if args[i] == nil {
+			return nil, errors.New("data is nil")
+		}
+		var front bool
+		var temp [][]string
+		switch args[i].TypeName() {
+		case indicators.AverageTrueRange:
+			temp, err = convertATR(args[i])
+		case indicators.BollingerBands:
+			temp, err = convertBollingerBands(args[i])
+		case indicators.ExponentialMovingAverage:
+			temp, err = convertEMA(args[i])
+		case indicators.MovingAverageConvergenceDivergence:
+			temp, err = convertMACD(args[i])
+		case indicators.MoneyFlowIndex:
+			temp, err = convertMFI(args[i])
+		case indicators.OnBalanceVolume:
+			temp, err = convertOBV(args[i])
+		case indicators.RelativeStrengthIndex:
+			temp, err = convertRSI(args[i])
+		case indicators.SimpleMovingAverage:
+			temp, err = convertSMA(args[i])
+		case indicators.OHLCV:
+			temp, err = convertOHLCV(args[i])
+			front = true
+		case "string":
+			var ok bool
+			target, ok = objects.ToString(args[i])
+			if !ok {
+				return nil, errors.New("failed to convert incoming output to string")
+			}
+		default:
+			err = fmt.Errorf("%s type is not handled", args[i].TypeName())
+		}
+		if err != nil {
+			return nil, err
+		}
+
+		if front {
+			var newBucket [][]string
+			newBucket = append(newBucket, temp...)
+			for x := range bucket {
+				newBucket[x] = append(newBucket[x], bucket[x]...)
+			}
+
+			bucket = newBucket
+			front = false
+			continue
+		}
+
+		if len(bucket) == 0 {
+			bucket = temp
+		} else {
+			for i := range temp {
+				bucket[i] = append(bucket[i], temp[i]...)
+			}
+		}
+	}
+	return nil, file.WriteAsCSV(target+".csv", bucket)
```

Comment:
Should we be prepending the "output" path here ?

For example if we call this from a script:
```go
    common.writeascsv("/Users/andrew/HELLOWORLD", ohlcvData, avgtr, simma, expma)
```

It will write to /Users/andrew/HELLOWORLD.csv

and if you leave it as output it will output to current execution path/output

Reply:
How would you like it?

Reply:
Unless i am mistaken based on the response from @thrasher- above and

https://github.com/thrasher-corp/gocryptotrader/pull/496/files#diff-3baf47c64847a8fb8aaa8cc2e088513bR1215-R1220

I thought the intent was to only allow scripts to output into ~/.gocryptotrader/scripts/output

I don't really mind either way but if this is no longer the case we can remove the creation of the output folder in checkGCTScriptConfig()

Reply:
Ok so I changed the way I handled it. It will always save in our scripts/output file. I also fixed a possible file transversal bug that I need you to check over. So now you can just pass in a ctx string which is defined when the script compiles or you can define the file yourself. You can actually change the file extension, so that file transversal might be bad, so let me know if you want this to always be a csv.